src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1628,42 +1628,60 @@
     (fs/delete csr-path)
     (log/debug (i18n/trs "Removed certificate request for {0} at ''{1}''" subject csr-path))))
 
+(schema/defn prevent-duplicate :- [schema/Int]
+  "Given a list of serials and Puppet's CA CRL, returns vector of serials with
+   any already-revoked serials removed."
+  [serials :- [schema/Int]
+   crl :- X509CRL]
+  (let [crl-revoked-list (.getRevokedCertificates crl)
+        existed-serials (set (map #(.getSerialNumber %) crl-revoked-list))
+        duplicate-serials (set/intersection (set serials) existed-serials)] ; For info output
+        (when (> (count duplicate-serials) 0)
+          (doseq [x duplicate-serials]
+            (log/info (str "Serial " x " existed in CRL.")))) ; Feedback on where to place the info output.
```

Comment:
We might want to put this just at the debug level? Not sure if users will want/need this info all the time.

I would also maybe say something like "Certificate with serial x has already been revoked." ?

Reply:
I actually kind of like this being at info. It'll help surface how often we try to do duplicate revocations, which is kind of an unknown right now.

Agreed about the rephrasing.